portfolio/src/main/webapp/task.js

```diff
@@ -6,20 +6,124 @@
+  const length = document.getElementById('new-task-estimated-length').value;
+  const lengthUnit =
+      document.getElementById('new-task-estimated-length-unit').value;
+  const priority = document.getElementById('new-task-priority').value;
+
+  const dataArray =
+      [name, description, getDuration(length, lengthUnit), priority];
+
+  if (name == '') {
+    return false;
+  }
+
+  if (parseInt(length) <= 0 || length == '') {
+    return false;
+  }
+
+  const newTask = new Task(
+      name, description, getDuration(length, lengthUnit), parseInt(priority));
+  updateTaskList(newTask);
+}
+
+/** Display Task information from user input. */
+function updateTaskList(newTask, lengthUnit) {
+  const newEventCard = document.createElement('div');
+  newEventCard.classList.add('card');
+
+  const cardBody = document.createElement('div');
+  cardBody.classList.add('card-body');
+  newEventCard.appendChild(cardBody);
+
+  const cardTitle = document.createElement('h4');
+  cardTitle.classList.add('card-title');
+  cardTitle.innerText = newTask.name;
+  cardBody.appendChild(cardTitle);
+
+  const descriptionText = document.createElement('p');
+  descriptionText.classList.add('card-text');
+  descriptionText.innerText = newTask.description;
+  cardBody.appendChild(descriptionText);
+
+  const duration = document.createElement('p');
+  duration.classList.add('card-text');
+  duration.innerText = newTask.duration;
+  cardBody.appendChild(duration);
+
+  const priority = document.createElement('p');
+  priority.classList.add('card-text');
+  priority.innerText = newTask.taskPriority;
+  cardBody.appendChild(priority);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('btn');
+  deleteButton.classList.add('btn-danger');
+  deleteButton.innerText = 'Delete this event';
+  cardBody.appendChild(deleteButton);
+
+  const eventList = document.getElementById('new-task-list');
+  eventList.innterHTML = '';
+  eventList.appendChild(newEventCard);
+
+  // The delete button removes the event's card from the UI.
+  deleteButton.onclick = function(newEventCard) {
+    newEventCard.target.closest('div.card').remove();
+  }
+}
+
+/** Return number of minutes from the user's input and unit selection. */
+function getDuration(duration, unit) {
+  if (unit == 'minutes') {
+    return String(duration);
+  } else if (unit == 'hours') {
+    return String(duration * 60);
+  }
+}
+
+/**
+ * Returns JSON data from an array of Javascript Task classes
+ * by gathering the information from the previously created
+ * cards used to display input Task information.
+ */
+function collectAllTasks() {
+  const allTaskJson = new Array();
+
+  // Get task list element to run through cards collecting data.
+  const taskList = document.getElementById('new-task-list');
+
+  taskList.childNodes.forEach((taskCard) => {
+    const taskName = taskCard.childNodes[0].childNodes[0].innerText;
```

Comment:
Like Tomas said on my PR, we can probably pull taskCard.childNodes[0] into a separate variable with a more descriptive name.

Reply:
Done.